src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,128 @@
+import { useContext } from 'react';
+import { PersonItem } from '../PersonItem';
+import { PeopleContext } from '../../peopleContext';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { SortBy, SortOrder, SortQueries } from '../../types/Sort';
+import { Person } from '../../types';
+import { sortPeople } from '../../utils/sortPeople';
+
+export const PeopleTable: React.FC = () => {
+  const [searchParams] = useSearchParams();
+  const filteredPeople = useContext(PeopleContext);
+  const navigate = useNavigate();
+
+  const sortByField = searchParams.get(SortQueries.sort);
+  const sortOrder = searchParams.get(SortQueries.order) || SortOrder.off;
+
+  if (!filteredPeople.length) {
+    return;
+  }
+
+  const sortHandler = (sortField: SortBy) => {
+    let tempSortOrder = SortOrder.off;
+
+    if (sortField === sortByField) {
+      switch (sortOrder) {
+        case SortOrder.off:
+          tempSortOrder = SortOrder.asc;
+          break;
+        case SortOrder.asc:
+          tempSortOrder = SortOrder.decs;
+          break;
+      }
+    } else {
+      tempSortOrder = SortOrder.asc;
+    }
+
+    const newParams = new URLSearchParams(searchParams);
+
+    newParams.set(SortQueries.sort, sortField);
+    newParams.set(SortQueries.order, tempSortOrder);
+
+    navigate({
+      search: newParams.toString(),
+    });
+  };
```

Comment:
The sortHandler function is quite complex. You could simplify it by using a ternary operator and removing the switch statement.